.devcontainer/pre-build.sh

```diff
@@ -21,6 +21,7 @@ sudo apt install -y protobuf-compiler
 
 # Make sure go path is owned by vscode
 sudo chown vscode:vscode /home/vscode/go || true
+sudo chown -R vscode:vscode /home/vscode/go/src || true
```

Comment:
```suggestion
sudo chown vscode:vscode /home/vscode/go/src || true
```

let me spin up a Codespace to make sure, but I think removing the `-R` worked fine for `make codegen`

Reply:
```suggestion
sudo chown vscode:vscode /home/vscode/go/src || true
sudo chown vscode:vscode /home/vscode/go/src/github.com || true
```

Had to add one more directory down in the Codespace. I'm not sure if only the last one is required or if all 3 are required. Might check that one more time

Reply:
ok [looks like](https://github.com/agilgur5/argo-workflows/commit/2e771049af3a1921ded47234eae3e28e9b918ba4) all 3 dirs are necessary but no more than those 😅

Reply:
I'm gonna make this change to unblock new devcontainers, seemed like you wouldn't mind per previous comments, but I [wasn't sure](https://github.com/argoproj/argo-workflows/pull/13270#issuecomment-2243999188)